src/Compilers/Core/Portable/System/Reflection/Metadata/Ecma335/MetadataBuilder.Heaps.cs

```diff
@@ -221,82 +221,43 @@ public int GetHeapOffset(UserStringHandle handle)
         }
 
         /// <exception cref="ImageFormatLimitationException">The remaining space on the heap is too small to fit the string.</exception>
+        public UserStringHandle ReserveUserString(int length, out Blob fixup)
+        {
+            int offset = GetNewUserStringHeapOffset(length);
+            int encodedLength = BlobUtilities.GetUserStringByteLength(length);
+            fixup = _userStringWriter.ReserveBytes(BlobWriterImpl.GetCompressedIntegerSize(encodedLength) + encodedLength);
```

Comment:
Good point. The caller should zero out the blob. The same for the call to ReserveGuid.

---
In reply to: [63581822](https://github.com/dotnet/roslyn/pull/11357#discussion_r63581822) [](ancestors = 63581822)